### ISSUE TITLE
If you try to output a playlist in HLS-V4 format, an error will occur.

### ISSUE THREAD

**Issue by HigashidaTomoyuki** [kuu/hls-parser#38]
When trying to output in HLS-V4 format, executing the following code causes an error.

```
const HLS = require('hls-parser');

const { MediaPlaylist, Segment } = HLS.types;
const segments = [];
Items.forEach((item) => {
  segments.push(
    new Segment({
      uri: item.uri,
      duration: item.duration,
      byterange: {
        length: item.byterangeLength,
        offset: item.byterangeOffset
      },
      mediaSequenceNumber: item.mediaSequenceNumber,
      discontinuitySequence: 0
    })
  )
});

const playlist = new MediaPlaylist({
  version: 4,
  targetDuration: 10,
  mediaSequenceBase: 0,
  playlistType: 'VOD',
  segments: segments
});

console.log(HLS.stringify(playlist))
```

-----error log-----

```
{
    "errorMessage": "Invalid Playlist : Redundant item (test01.ts)",
    "errorType": "Error",
    "stackTrace": [
        "Object.INVALIDPLAYLIST (/var/task/node_modules/hls-parser/utils.js:44:9)",
        "LineArray.push (/var/task/node_modules/hls-parser/stringify.js:34:15)",
        "buildSegment (/var/task/node_modules/hls-parser/stringify.js:218:9)",
        "buildMediaPlaylist (/var/task/node_modules/hls-parser/stringify.js:191:5)",
        "Object.stringify (/var/task/node_modules/hls-parser/stringify.js:280:5)",
        "module.exports.handler (/var/task/index.js:89:19)",
        "<anonymous>",
        "process._tickDomainCallback (internal/process/next_tick.js:228:7)"
    ]
}
```

I do not know if coding is bad or hls-parser is faulty. Please tell me how to resolve.
If byterange is not written, no error occurs and it works normally.

**Comment by kuu** [comment]
Assuming the array `Items` contains redundant URLs ("test01.ts")
Can you provide more detail about what you are trying to do?
For what use-cases do you need the redundant segments?

**Comment by HigashidaTomoyuki** [comment]
The use case uploads the video by streaming, and it is trying to add it to the playlist so that it can be played from the uploading completion. Therefore, the `Item` array stores URLs of playable ts files, `Byterange`, and so on.

**Comment by kuu** [comment]
Can you provide the code you used for initializing `Item` variable?

**Comment by kuu** [comment]
I mean, can you provide the data dump of the `Item` variable?

**Comment by HigashidaTomoyuki** [comment]
An example of substitution value is here.

```
Items = [{ 
  byterangeOffset: 0,
  trackId: '001',
  id: '001',
  byterangeLength: 115244,
  uri: 'test01.ts',
  duration: 10.048,
  mediaSequenceNumber: 0 
},{ 
  byterangeOffset: 115244,
  trackId: '002',
  id: '001',
  byterangeLength: 115620,
  uri: 'test01.ts',
  duration: 9.984,
  mediaSequenceNumber: 1
}]
```

**Comment by kuu** [comment]
Thanks for reporting this.
I think I fixed the bug by https://github.com/kuu/hls-parser/pull/39 and it's already published to npm as hls-parser@0.2.4
Can you doublecheck?
